docs/sdk-configuration.md

```diff
@@ -0,0 +1,492 @@
+      .setClock(AtomicClock.create())
+      .setPropagators(B3Propagator.getInstance())
+      .addSpanExporter(OtlpGrpcSpanExporter.builder()
+        .setEndpoint(config.getOtlpExporter().getEndpoint())
+        .setTimeout(config.getOtlpExporter().getTimeout())
+        .setBatchQueueSize(config.getOtlpExporter().getQueueSize())
+        .build())
+      .addSpanExporter(ZipkinExporter.builder()
+        .setEndpoint(config.getZipkinExporter().getEndpoint())
+        .disableBatching()
+        .build())
+      .addSpanExporter(FullyAsyncZipkinExporter.create())
+      .addSpanProcessor(CustomAttributeAddingProcessor.create())
+      .addSpanProcessor(CustomEventAddingProcessor.create())
+      .setTraceSampler(config.getSamplingRate() != 0 
+          ? Sampler.rateLimiting(config.getSamplingRate()) : Sampler.getDefault())
+      .setTraceLimits(TraceLimits.builder().setMaxAttributes(config.getMaxSpanAttributes()).build())
+      .setIdGenerator(TimestampedIdGenerator.create())
+      .setMeterStuff(InProgressMetricsStuff.create())
+      .build();
+  }
+  
+  @Bean
+  public TracerProvider tracerProvider(OpenTelemetry openTelemetry) {
+    return openTelemetry.getTracerProvider();
+  }
+  
+  @Bean
+  public MeterProvider meterProvider(OpenTelemetry openTelemetry) {
+    return openTelemetry.getMeterProvider();
+  }
+  
+  @Bean
+  public AuthServiceStub authService(OpenTelemetry openTelemetry, AuthConfig config) {
+    return AuthServiceGrpc.newBlockingStub(ManagedChannelBuilder.forEndpoint(config.getEndpoint()))
+      .withInterceptor(TracingClientInterceptor.create(openTelemetry));
+  }
+  
+  @Bean
+  public ServletFilter servletFilter(OpenTelemetry openTelemetry) {
+    return TracingServletFilter.create(openTelemetry);
+  }
+}
+
+// Use some instrumented client
+@Component
+public class MyAuthInterceptor {
+
+  private final AuthServiceStub authService;
+  
+  @Inject
+  public MyAuthInterceptor(AuthServiceStub authService) {
+    this.authService = authService;
+  }
+  
+  public void doAuth() {
+    if (!authService.getToken("credential").isAuthenticated()) {
+      throw new HackerException();
+    }
+  }
+}
+
+// Use tracer directly, not so common
+@Component
+public class MyService {
+
+  private final Tracer tracer;
+  private final Meter meter;
+  
+  @Inject
+  public MyService(TracerProvider tracerProvider, MeterProvider meterProvider) {
+    tracer = tracerProvider.get("my-service");
+    meter = meterProvider.get("my-service");
+  }
+  
+  public void doLogic() {
+    Span span = tracer.spanBuilder("logic").startSpan();
+    try (Scope ignored = span.makeCurrent()) {
+      Thread.sleep(1000);
+    } finally {
+      span.end();
+    }
+  }
+}
+```
+
+## The global instance of the SDK
+
+A built instance is convenient to use in most Java apps because of dependency injection. Because it
+has a easy-to-reason lifecycle, being tied into the dependency ordering (even if dependency 
```

Comment:
I agree with this section, but not with "easy-to-reason lifecycle". Shared instance kept in a field will break if somebody somewhere calls `shutdown` on the same shared instance.